Landing Page and related files

psarkar2018 opened the pull request.

akdombrowski commented on client/package.json:
```diff
@@ -3,15 +3,16 @@
     "redux": "^4.0.1",
-    "redux-thunk": "^2.3.0",
-    "axios": "0.18.0"
+    "redux-thunk": "^2.3.0"
```
Nit. Looks like axios was moved up a few lines. I think the only real change is adding react-router-dom. Update the file so we just see the change to this package?

eeeeeric replied:
npm automatically reorganizes the dependencies to alphabetical order when you do `npm install --save pkg`.

psarkar2018 replied:
I don't mind moving axios back, but should we keep it this way in case some one has to install another package in the future and it gets reshuffled again?

eeeeeric replied:
I think we should just keep it as is (so no changes from the PR).

psarkar2018 replied:
Present revision?

akdombrowski commented on client/src/styles/LandingPage.css:
```diff
@@ -0,0 +1,71 @@
+    width: 100vw;
+    margin: auto;
+    background: url("../components/images/lotus.jpg") no-repeat 50% 50%;
+    display: table;
```
Reactstrap has easy compatibility with flexbox, and I've seen that tables can do their own thing sometimes. Would it be difficult to convert this to use flexbox instead?

psarkar2018 replied:
XD For me it might be a difficult, but I don't mind looking into it if we need it! I sent you a demo of the landing page, tho, and its pretty responsive so maybe we don't need flexbox?

psarkar2018 replied:
I updated the landing page to flexbox! It was easier than I thought it would be!

psarkar2018 commented:
I'm not sure why, but when I tried to run `npm start` in the client directory when I first forked the mystic-get-together repo, I got some compile errors, so I added those lines to the sdk files (my bad about the newlines XD). I'm guessing you don't get any compile errors when running `npm start` in the client directory?

eeeeeric commented:
@psarkar2018 You might need to merge the latest changes from master into your branch.

psarkar2018 commented:
@eeeeeric I think I have the latest changes from warlocks-of-the-midwest:master, tho, because I forked from it, and there hasn’t new changes to it since I forked. 

I wonder if we’re using different versions of javascript or something? For example, I would get errors in sdk.js until I moved the db variable in line 1 outside the curly braces, which seems to be an ES6 syntax (https://stackoverflow.com/questions/36795819/when-should-i-use-curly-braces-for-es6-import/36796281), but the version in master keeps it inside the braces. 
Also I would get errors in sdk.js line 1 until I added .js extension to the ‘./fire’, which seem sot be another ES6 syntax (https://stackoverflow.com/questions/44481851/does-es6-import-export-need-js-extension) but the version in master didn’t need that.

eeeeeric commented:
@psarkar2018 I wanted to see if I had the same problem but I am blocked by some error when running `npm install` right now. Maybe this is due to a mismatch in Node versions between you and Henry/Jake. Maybe they can chime in @hrecker @jtenberg

hrecker commented:
Hmm I have no npm error on my InitialJsSDK branch where I added the SDK, but in master where that was merged I do have the error. I'll look into it more

hrecker commented:
It looks like master was able to work until the "Provide methods in the JS SDK to call the Cloud Functions." commit. I think if these commits fix the problem we can just merge them.
